File: critique/test/scripts/version.test.ts
/**
 * @jest-environment node
 */
describe('critique version script', () => {
  it('should generate a tag with the correct format', async () => {
    const capability = process.cwd().split('/').pop();
    const version = await import('../../package.json');
    const tag = `${capability}-v${version.version}`;

    expect(version.version.split('.').length).toBe(3);
    expect(tag.startsWith(`${capability}-v`)).toBe(true);
  });
});

File: ideate/tests/scripts/version_test.py
"""
Test ideate version script for correct tag naming.
"""
import os
import subprocess


def test_version_script_tag_format() -> None:
    """
    Test that the version script would create the correct tag format.
    """
    result = subprocess.run(
        ["python3", "-c", "from __version__ import __version__; print(__version__)"],
        capture_output=True, text=True, check=False
    )

    capability = os.getcwd().split("/").pop()
    version = result.stdout.strip()
    tag = f"{capability}-v{version}"

    assert tag.startswith("ideate-v")
    assert tag.count('.') == 2
